pkg/controllers/oauthclientscontroller/oauthclientscontroller.go

```diff
@@ -92,13 +101,12 @@ func (c *oauthsClientsController) getIngressConfig() (*configv1.Ingress, error)
 	return ingress, nil
 }
 
-func (c *oauthsClientsController) getCanonicalRouteHost(ingressConfigDomain string) (string, error) {
+func (c *oauthsClientsController) getCanonicalRouteHost(expectedHost string) (string, error) {
```

Comment:
will this work? IIRC `routeapihelpers.IngressURI` checks for admitted ingresses, will `expectedHost` be among them if it's outside the router's domain?